### PR TITLE
Update EvalML section of tutorials 

### LINKED ISSUE #197: Predict Next Purchase Tutorial: evalml split_data and problem formation
Hi, I have very much enjoyed seeing the progress that the alteryx team is making on these projects, very exciting and impressive stuff!

I was going through the [predict next purchase tutorial](https://render.githubusercontent.com/view/ipynb?color_mode=light&commit=3a83a05c1e3595a0d929b505dfecb273020780ac&enc_url=68747470733a2f2f7261772e67697468756275736572636f6e74656e742e636f6d2f616c74657279782f636f6d706f73652f336138336130356331653335393561306439323962353035646665636232373330323037383061632f646f63732f736f757263652f6578616d706c65732f707265646963745f6e6578745f70757263686173652e6970796e62&nwo=alteryx%2Fcompose&path=docs%2Fsource%2Fexamples%2Fpredict_next_purchase.ipynb&repository_id=163425829&repository_type=Repository#Splitting-the-Data) , and I ran into a couple of small issue, and also had a question about the modeling approach...

It looks like the EvalML API has been updated since this was created

Upon running 

```
y = fm.pop('bought_product')

splits = evalml.preprocessing.split_data(
    X=fm,
    y=y,
    test_size=0.2,
    random_state=0,
)

X_train, X_holdout, y_train, y_holdout = splits
```

I hit `TypeError: split_data() missing 1 required positional argument: 'problem_type'`

- adding `problem_type='binary'` as a kwarg param fixed that

And again, in running

```
automl = evalml.AutoMLSearch(
    problem_type='binary',
    objective='f1',
    random_state=0,
    allowed_model_families=['catboost', 'random_forest'],
    max_iterations=3,
)
```

I see `ValueError: Must specify training data as a 2d array using the X_train argument`

which, I assume should be 
```
automl = evalml.AutoMLSearch(
    X_train=X_train,
    y_train=y_train,
    problem_type='binary',
    objective='f1',
    random_state=0,
    allowed_model_families=['catboost', 'random_forest'],
    max_iterations=3,
)
```
Looks like the other tutorials need to be updated too.

I'm happy to make these changes, assuming there isn't a crazy docs build process or something. But, should this tutorial be added as some sort of automated test, and/or evalml be pinned in test_requirements.txt?

Now, about the modeling approach...I see that is being modeled as a simple 'binary' problem. This may be naive, but I wondered if it made any sense to model it as a time series classification problem, instead (perhaps even a case of https://github.com/alteryx/evalml/issues/1506)? Some of the customers in this data set are repeats, and thus show up in multiple rows, making those rows _not_ independent of each other, I think. While some customers only show up once, as a single row.

From the tutorial - user 6851 shows up once, and user 8167 shows up twice:
Screen Shot 2021-01-10 at 1 16 38 AM

The turbofan degredation example seems to be even more "time series" oriented.

Some sort of k-fold cross-validation and/or a test/train split that doesn't take those repeat customers is what I would be most worried about. Mostly, just curious what your team's thoughts are about this and if there's a clear way that y'all make this kind of distinction.

 Thanks

### PR DESCRIPTION
**Pull request by flowersw**
This simply updates the evalml section of the tutorial notebooks in the documentation, to be up to date with evalml 0.17, and pins evalml and woodwork higher and gte in the docs requirements.txt file.

I did want to note that in case these tutorials are benchmark or meant to be exactly reproducible in some way, I noticed a few slight differences in the output of the models, e.g. slightly different evaluation accuracy, slightly different feature importances, etc.. Unless I'm missing a new "randomness" parameter, I would guess that's due to the underlying changes in evalml, but am not sure.

### CODE REVIEW

**Review by jeff-hernandez — COMMENTED** [review]
Looking good! Lets clear the outputs in the notebooks so the build process can automatically run the notebooks.

File: docs/requirements.txt
Comment (flowersw) [reply docs/requirements.txt]: sounds good

File: docs/source/examples/predict_turbofan_degredation.ipynb
Comment (jeff-hernandez) [inline docs/source/examples/predict_turbofan_degredation.ipynb:374]: For consistency, lets also use keyword arguments here.

**Review by jeff-hernandez — COMMENTED** [review]
This looks great! I left one minor comment. Other than that, we'll be ready to merge after updating with main.

File: docs/source/examples/predict_turbofan_degredation.ipynb
Comment (flowersw) [reply docs/source/examples/predict_turbofan_degredation.ipynb]: good catch

**Comment by flowersw** [comment]
happy to squash or rebase if y'all do that, too

**Review by jeff-hernandez — APPROVED** [review]
Looks good! If you want to automatically close issue #197 after merging this PR, you can link it by using a supported keyword in the pull request's description -- more info [here](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).  Thank you for the contribution @flowersw!

**Comment by flowersw** [comment]
Fixes alteryx/compose#197

**Comment by flowersw** [comment]
Ok, I think we're good. I'm assuming you'll merge it, as I don't seem to be able to. Thanks

**Comment by dsherry** [comment]
Thanks @flowersw ! Yes, the difference in output you saw is due to [updates we've made to EvalML](https://evalml.alteryx.com/en/stable/release_notes.html), including adding more models and fixing bugs.

**Pull request merged**
